### ISSUE TITLE
Robot pick fails

### ISSUE THREAD

**Issue by jrgnicho** [swri-robotics/Gilbreth#35]
When running the main application (see [Demo.md](https://github.com/swri-robotics/gilbreth/blob/kinetic-devel/DEMO.md)), as a part comes close to the robot, the robot arm moves in position to pick the object, then it lowers its tool towards the conveyor just in time to touch the target object.  Once the tool and object make contact the gripper appears to fail at making a successful grasp.  After re-attempting a grasp for approximately 10 seconds the robot gives up and retreats.

**Comment by alex07zzz** [comment]
Yes, I also found this problem. And sometimes the object is attached to the side of the gripper, but the gripper attached state  will still be false. The vacuum gripper seems to have a very strict z-direction height and orientation requirement.

I noticed that we have a very complex-looking vacuum gripper. Could it be the reason that the collision size of our gripper's tip is too small? I didn't notice the same issue for ARIAC gripper which is big and flat.



在 2018年2月1日，下午3:46，Jorge Nicho <notifications@github.com<mailto:notifications@github.com>> 写道：


When running the main application (see Demo.md<https://nam01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fswri-robotics%2Fgilbreth%2Fblob%2Fkinetic-devel%2FDEMO.md&data=02%7C01%7C%7C051c030921ea409673f108d569ce1348%7C84df9e7fe9f640afb435aaaaaaaaaaaa%7C1%7C0%7C636531256157508403&sdata=bmEA97xHxLWTZy4VK3kSaY6iLWfq2Wg%2B7xBwYx4Hix4%3D&reserved=0>), as a part comes close to the robot, the robot arm moves in position to pick the object, then it lowers its tool towards the conveyor just in time to touch the target object. Once the tool and object make contact the gripper appears to fail at making a successful grasp. After re-attempting a grasp for approximately 10 seconds the robot gives up and retreats.

**Comment by jrgnicho** [comment]
After observing the robot more carefully, it looks like the robot tool does not make contact with the top surface of the object as it is going by, it instead arrives at the conveyor a bit too early and blocks the object from moving forward.  That pick strategy needs to be modified.  I'd suggest moving the gripper into position just a couple of cm above the expected pick point and then wait until the object passes right underneath it; at this point the arm will move the tool downwards towards the pick point and turn on the gripper.  This will require implementing a countdown logic in the robot execution node that will execute that move at the right time. Also, the rail or the arm should be moved one at a time and never simultaneously.  
The tool size may be an issue too but our grip strategy will continue to be an problem even if we modify the tool.

**Comment by alex07zzz** [comment]
I agree with you. I will change the strategy.

在 2018年2月2日，上午8:32，Jorge Nicho <notifications@github.com<mailto:notifications@github.com>> 写道：


After observing the robot more carefully, it looks like the robot tool does not make contact with the top surface of the object as it is going by, it instead arrives at the conveyor a bit too early and blocks the object from moving forward. That pick strategy needs to be modified. I'd suggest moving the gripper into position just a couple of cm above the expected pick point and then wait until the object passes right underneath it; at this point the arm will move the tool downwards towards the pick point and turn on the gripper. This will require implementing a countdown logic in the robot execution node that will execute that move at the right time. Also, the rail or the arm should be moved one at a time and never simultaneously.
The tool size may be an issue too but our grip strategy will continue to be an issue even if we modify the tool.

**Comment by marianotorrez** [comment]
ATTENTION ARE YOU IN NEED OF PERSONAL LOAN/INVESTMENT LOAN/BUSINESS FUNDING???Email: marianotorrezloans@gmail.com

 We have provided over $30 Billion in business loans to over 50,000 business owners just like you. We use our own designated risk technology to provide you with the right business loan so you can grow your business. Our services are fast and reliable, loans are approved within 24 hours of successful application. Do you find yourself in a bit of trouble with unpaid bills and don't know which way to go or where to turn? What about finding a reputable Debt Consolidation firm that can assist you in reducing monthly installment so that you will have affordable repayment options as well as room to breathe when it comes to the end of the month and bills need to get paid? LoansNow international financial company is the answer. Reduce your payments to ease the strain on your monthly expenses. We offer loans from a minimum range of $10,000 to a maximum of $700 million Dollars. Intermediaries/Consultants/Brokers are welcome to bring their clients and are 100% protected. Give us a try and be free from financial problems by Phone Contact: +1 (405)383-3602 Email: marianotorrezloans@gmail.com In complete confidence, we will work together for the benefits of all parties involved. Do not keep your financial problems to yourself in order for you not to be debt master or financial stress up, which is why you must contact us quickly for a solution to your financial problems. It will be a great joy to us when you are financially stable Email us via: marianotorrezloans@gmail.com

   Our services include the following below
______________________________________________________
*Personal loans *Truck & Car Loans  *Debt consolidation loans
*Student Loans  *Home & Hotel Loans *Marriage & Celebration Loans 
*Agricultural & Farm Loans *Real Estate & Construction Loans
*Business Start-up Loans or Business Expansion Loans. We offer all types of loans

  We are a financial consultants that handles international finances for any amount of banking instruments. We have the access/contacts to raise from $10 thousand to $700 Million Dollars. We have brought ailing industries back to life and we back good business ideas by providing funds for their up start. We have network of Investors that are willing to provide funds of whatever amount to individuals and organizations to start business and operations. Our company has recorded a lot of breakthroughs in the provision of first class financial services to our clients, especially in the area of Loan syndication and capital provision for individuals and companies. We offer loan to individual and public sector that are in need of financial Assistance in a low interest rate of 2%. Bad credit acceptable,The Terms and Conditions are very simple and considerate, We are a group of energetic and experienced loan professionals with thorough knowledge of financial markets, we are committed in providing our customers and suppliers with simple and competitive forms of finance solutions for all business users. I believe you might be interested in starting up a new business in need of funding or expanding the scope of your already existing businesses, our investors are also interested in funding / partnering and ready to invest large sum on traders who are ready to proof their expertise in the new industry with our reliable loan company and other TOP PRIME AAA BANKS like Bank of America, HSBC, Lloyds, Wells Fargo, The Best Banks for SBA Loans and many more. You will never regret anything in this loan transaction because i will make you happy via email {marianotorrezloans@gmail.com} getting a loan from this company is 100% assured and guaranteed contact by email marianotorrezloans@gmail.com

Regards
Dr. Mariano Torrez 
Phone Contact: +1 (405)383-3602
Email: marianotorrezloans@gmail.com
